.github/workflows/ci.yaml

```diff
@@ -806,18 +806,94 @@ jobs:
 
           echo "Required checks have passed"
 
+  # Builds the dylibs and upload it as an artifact so it can be embedded in the main build
+  build-dylib:
+    needs: changes
+    # TODO: Uncomment
+    # if: github.ref == 'refs/heads/main' && needs.changes.outputs.docs-only == 'false' && !github.event.pull_request.head.repo.fork
+    runs-on: ${{ github.repository_owner == 'coder' && 'depot-macos-latest' || 'macos-latest' }}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+        with:
+          egress-policy: audit
+
+      - name: Checkout
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 0
+
+      - name: Setup build tools
+        run: |
+          brew install bash gnu-getopt make
+          echo "$(brew --prefix bash)/bin" >> $GITHUB_PATH
+          echo "$(brew --prefix gnu-getopt)/bin" >> $GITHUB_PATH
+          echo "$(brew --prefix make)/libexec/gnubin" >> $GITHUB_PATH
+
+      - name: Setup Go
+        uses: ./.github/actions/setup-go
+
+      - name: Install rcodesign
```

Comment:
We shouldn't need to sign artifacts built for PRs since they aren't being deployed, this is just for validating that the code builds properly

Reply:
The macOS Network Extension won't load the dylib unless it's signed. If we don't sign it and run Coder Desktop pointing at `dev.coder.com`, it'd download a dylib it can't use.

Reply:
This is just to sign the dylib for builds off `main` btw, not PRs

Reply:
OK, but we probably need another job that verifies everything builds then on Go changes so we dont merge unbuildable code

Reply:
Could we just make `build-dylib` run on PRs too?

Reply:
Oh and then we're signing it for no reason, I see

Reply:
Yeah I would like that, but what I'm trying to say is that it shouldn't bother signing it. We should probably avoid pulling down a cert secret into PR builds if it's unnecessary.

Reply:
I'm just gonna skip the signing steps on main and then have 
`CODER_SIGN_DARWIN: ${{ github.ref == 'refs/heads/main' && '1' || '0' }}`